linkerd/app/outbound/src/http/detect.rs

```diff
@@ -54,48 +46,23 @@ impl<T> Outbound<T> {
             .push(http::NewServeHttp::layer(h2_settings, rt.drain.clone()))
             .push_map_target(http::Logical::from)
             .push(svc::UnwrapOr::layer(
-                // When an HTTP version cannot be detected, we fallback to a logical
-                // TCP stack. This service needs to be buffered so that it can be
-                // cached and cloned per connection.
-                svc::stack(tcp.clone())
-                    .push(profiles::split::layer())
-                    .push_switch(ShouldResolve, tcp_forward.clone())
-                    .push_on_response(
-                        svc::layers()
-                            .push_map_target(io::EitherIo::Right)
-                            .push(rt.metrics.stack.layer(stack_labels("tcp", "logical")))
-                            .push(svc::layer::mk(svc::SpawnReady::new))
-                            .push(svc::FailFast::layer("TCP Logical", dispatch_timeout))
-                            .push_spawn_buffer(buffer_capacity),
-                    )
-                    .instrument(|_: &_| debug_span!("tcp"))
-                    .check_new_service::<tcp::Logical, _>()
+                tcp.clone()
+                    .push_on_response(svc::layers().push_map_target(io::EitherIo::Right))
```

Comment:
nit/TIOLI: could just be this, right?
```suggestion
                    .push_on_response(svc::MapTargetLayer::new(io::EitherIo::Right))
```